crates/storage_impl/src/payouts/payout_attempt.rs

```diff
@@ -0,0 +1,478 @@
+    DataModelExt, DatabaseStore, KVRouterStore,
+};
+
+#[async_trait::async_trait]
+impl<T: DatabaseStore> PayoutAttemptInterface for KVRouterStore<T> {
+    #[instrument(skip_all)]
+    async fn insert_payout_attempt(
+        &self,
+        new_payout_attempt: PayoutAttemptNew,
+        storage_scheme: MerchantStorageScheme,
+    ) -> error_stack::Result<PayoutAttempt, StorageError> {
+        match storage_scheme {
+            MerchantStorageScheme::PostgresOnly => {
+                self.router_store
+                    .insert_payout_attempt(new_payout_attempt, storage_scheme)
+                    .await
+            }
+            MerchantStorageScheme::RedisKv => {
+                let key = format!(
+                    "mid_{}_poa_{}",
+                    new_payout_attempt.merchant_id, new_payout_attempt.payout_id
+                );
+                let field = format!("poa_{}", new_payout_attempt.payout_id);
+                let now = common_utils::date_time::now();
+                let created_attempt = PayoutAttempt {
+                    payout_attempt_id: new_payout_attempt.payout_attempt_id.clone(),
+                    payout_id: new_payout_attempt.payout_id.clone(),
+                    customer_id: new_payout_attempt.customer_id.clone(),
+                    merchant_id: new_payout_attempt.merchant_id.clone(),
+                    address_id: new_payout_attempt.address_id.clone(),
+                    connector: new_payout_attempt.connector.clone(),
+                    connector_payout_id: new_payout_attempt.connector_payout_id.clone(),
+                    payout_token: new_payout_attempt.payout_token.clone(),
+                    status: new_payout_attempt.status,
+                    is_eligible: new_payout_attempt.is_eligible,
+                    error_message: new_payout_attempt.error_message.clone(),
+                    error_code: new_payout_attempt.error_code.clone(),
+                    business_country: new_payout_attempt.business_country,
+                    business_label: new_payout_attempt.business_label.clone(),
+                    created_at: new_payout_attempt.created_at.unwrap_or(now),
+                    last_modified_at: new_payout_attempt.last_modified_at.unwrap_or(now),
+                    profile_id: new_payout_attempt.profile_id.clone(),
+                    merchant_connector_id: new_payout_attempt.merchant_connector_id.clone(),
+                    routing_info: new_payout_attempt.routing_info.clone(),
+                };
+
+                let redis_entry = kv::TypedSql {
+                    op: kv::DBOperation::Insert {
+                        insertable: kv::Insertable::PayoutAttempt(
+                            new_payout_attempt.to_storage_model(),
+                        ),
+                    },
+                };
+
+                // Reverse lookup for payout_attempt_id
+                let reverse_lookup = ReverseLookupNew {
```

Comment:
would reverse lookup be needed here, find functions dont seem to have a lookup for this